src/test/init/features/functions.spec.ts

```diff
@@ -0,0 +1,236 @@
+import * as sinon from "sinon";
+import { Config } from "../../../config";
+import * as prompt from "../../../prompt";
+import { Setup } from "../../../init";
+import { doSetup } from "../../../init/features/functions";
+import { Options } from "../../../options";
+import { RC } from "../../../rc";
+import { expect } from "chai";
+import * as _ from "lodash";
+
+function createExistingTestSetupAndConfig(): { setup: Setup; config: Config } {
+  const cbconfig = {
+    source: "testsource",
+    codebase: "testcodebase",
+    ignore: ["node_modules", ".git", "firebase-debug.log", "firebase-debug.*.log"],
+    predeploy: ['npm --prefix "$RESOURCE_DIR" run lint'],
+  };
+
+  return {
+    setup: {
+      config: {
+        functions: [cbconfig],
+      },
+      rcfile: { projects: {} },
+      featureArg: true,
+    },
+    config: new Config(
+      {
+        functions: [cbconfig],
+      },
+      { projectDir: "test", cwd: "test" }
+    ),
+  };
+}
+
+describe("functions", () => {
+  const sandbox: sinon.SinonSandbox = sinon.createSandbox();
+  let promptOnceStub: sinon.SinonStub;
+  let promptStub: sinon.SinonStub;
+  // let spawnStub: sinon.SinonStub;
+  let askWriteProjectFileStub: sinon.SinonStub;
+  let emptyConfig: Config;
+  let options: Options;
+
+  beforeEach(() => {
+    promptOnceStub = sandbox.stub(prompt, "promptOnce").throws("Unexpected promptOnce call");
+    promptStub = sandbox.stub(prompt, "prompt").throws("Unexpected prompt call");
+    // spawnStub = sandbox.stub(spawn, "spawn");
+
+    emptyConfig = new Config("{}", {});
+    options = {
+      cwd: "",
+      configPath: "",
+      only: "",
+      except: "",
+      filteredTargets: [],
+      force: false,
+      json: false,
+      nonInteractive: false,
+      interactive: false,
+      debug: false,
+      config: emptyConfig,
+      rc: new RC(),
+    };
+  });
+
+  afterEach(() => {
+    sandbox.restore();
```

Comment:
```suggestion
    sandbox.verifyAndRestore();
```

To make sure mocks are verified/satisfied.